lib/util.js

```diff
@@ -715,6 +715,9 @@ exports._extend = function(origin, add) {
   // Don't do anything if add isn't an object
   if (add === null || typeof add !== 'object') return origin;
 
+  // Create new object for origin, if origin isn't an object
+  if (origin === null || typeof origin !== 'object') origin = {};
```

Comment:
I agree with @brendanashworth. `util._extend(36, {a: 42})` is nonsensical and should probably blow up.